src/lib.rs

```diff
@@ -615,6 +614,22 @@ impl<K: Hash + Eq + PartialOrd, V: PartialOrd, S: HashState> PartialOrd for Link
     fn partial_cmp(&self, other: &LinkedHashMap<K, V, S>) -> Option<Ordering> {
         iter::order::partial_cmp(self.iter(), other.iter())
     }
+
+    fn lt(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::lt(self.iter(), other.iter())
+    }
+
+    fn le(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::le(self.iter(), other.iter())
+    }
+
+    fn ge(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::ge(self.iter(), other.iter())
```

Comment:
They have the same performance characteristics modulo the underlying `{eq, ne, lt, le, ge, gt}` impls. I think collection types should do their best to honor those (and other specializations).